pkg/reduce/readloop/readloop.go

```diff
@@ -144,21 +144,26 @@ func (rl *ReadLoop) Process(ctx context.Context, messages []*isb.ReadMessage) {
 		rl.log.Errorw("Failed to write messages", zap.Int("totalMessages", len(messages)), zap.Int("writtenMessage", len(successfullyWrittenMessages)))
 	}
 
+	if len(successfullyWrittenMessages) == 0 {
```

Comment:
why will it be == 0? We have checked early in the reduce code to continue only if there are `Read` messages. What we read should have been written, right?

Reply:
yeah, removed the check